client/consensus/beefy/src/communication/gossip.rs

```diff
@@ -130,9 +171,81 @@ where
 	/// Update gossip validator filter.
 	///
 	/// Only votes for `set_id` and rounds `start <= round <= end` will be accepted.
-	pub(crate) fn update_filter(&self, filter: GossipVoteFilter<B>) {
+	pub(crate) fn update_filter(&self, filter: GossipFilter<B>) {
```

Comment:
This operation is a bit heavier now since the filter contains the `validator_set()`. Could we receive a reference to the `validator_set` here and clone it and update it only if the `set_id` changes maybe ?